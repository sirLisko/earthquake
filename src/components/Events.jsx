import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';

import theme from 'styles/theme';

import { getCurrentFilter } from 'reducers/filter';
import { filterEvents } from 'utils/events';

const Events = ({ events }) => {
  const filter = useSelector(getCurrentFilter);
  const filteredEvents = filterEvents(events, filter);
  if (filteredEvents.length === 0) {
    return <div>Not found!</div>;
  }
  return (
    <table style={{ border: `1px solid ${theme.colors.gray_light}` }}>
      <tbody>
        {filteredEvents.map(({ id, properties: { mag, place, magType } }) => (
          <tr key={id}>
            <td>{id}</td>
            <td>{place}</td>
            <td>{mag}</td>
            <td>{magType}</td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export default Events;

Events.propTypes = {
  events: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string }))
    .isRequired,
};
