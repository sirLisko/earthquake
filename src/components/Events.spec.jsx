import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';

import Events from './Events';

const mockStore = configureStore([]);

describe('Events Component', () => {
  let wrapper;
  const props = {
    events: [
      {
        id: '123',
        properties: { mag: '1', place: 'iAmAPlace', maagType: 'foo' },
      },
      {
        id: '321',
        properties: { mag: '01', place: 'foobar', maagType: 'bar' },
      },
    ],
  };
  const emptyStore = mockStore({ filter: {} });

  beforeEach(() => {
    wrapper = mount(
      <Provider store={emptyStore}>
        <Events {...props} />
      </Provider>,
    );
  });

  it('should render properly', () => {
    expect(wrapper.find('Events')).toMatchSnapshot();
    expect(wrapper.find('tr').length).toBe(2);
  });

  it('should render properly - if no events', () => {
    wrapper = mount(
      <Provider store={emptyStore}>
        <Events events={[]} />
      </Provider>,
    );
    expect(wrapper.find('Events')).toMatchSnapshot();
    expect(wrapper.find('tr').length).toBe(0);
  });

  it('should render properly - with filters', () => {
    const emptyStore = mockStore({
      filter: { name: 'place', value: 'foobar' },
    });
    wrapper = mount(
      <Provider store={emptyStore}>
        <Events {...props} />
      </Provider>,
    );
    expect(wrapper.find('Events')).toMatchSnapshot();
    expect(wrapper.find('tr').length).toBe(1);
  });
});
