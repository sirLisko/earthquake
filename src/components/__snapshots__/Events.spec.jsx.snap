// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Component should render properly - if no events 1`] = `
<Events
  events={Array []}
>
  <div>
    Not found!
  </div>
</Events>
`;

exports[`Events Component should render properly - with filters 1`] = `
<Events
  events={
    Array [
      Object {
        "id": "123",
        "properties": Object {
          "maagType": "foo",
          "mag": "1",
          "place": "iAmAPlace",
        },
      },
      Object {
        "id": "321",
        "properties": Object {
          "maagType": "bar",
          "mag": "01",
          "place": "foobar",
        },
      },
    ]
  }
>
  <table
    style={
      Object {
        "border": "1px solid rgb(239, 239, 239)",
      }
    }
  >
    <tbody>
      <tr
        key="321"
      >
        <td>
          321
        </td>
        <td>
          foobar
        </td>
        <td>
          01
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</Events>
`;

exports[`Events Component should render properly 1`] = `
<Events
  events={
    Array [
      Object {
        "id": "123",
        "properties": Object {
          "maagType": "foo",
          "mag": "1",
          "place": "iAmAPlace",
        },
      },
      Object {
        "id": "321",
        "properties": Object {
          "maagType": "bar",
          "mag": "01",
          "place": "foobar",
        },
      },
    ]
  }
>
  <table
    style={
      Object {
        "border": "1px solid rgb(239, 239, 239)",
      }
    }
  >
    <tbody>
      <tr
        key="123"
      >
        <td>
          123
        </td>
        <td>
          iAmAPlace
        </td>
        <td>
          1
        </td>
        <td />
      </tr>
      <tr
        key="321"
      >
        <td>
          321
        </td>
        <td>
          foobar
        </td>
        <td>
          01
        </td>
        <td />
      </tr>
    </tbody>
  </table>
</Events>
`;
